src/python/pants/backend/python/lint/bandit/rules.py

```diff
@@ -68,10 +68,7 @@ async def bandit_lint_partition(
             output_filename="bandit.pex",
             internal_only=True,
             requirements=PexRequirements(bandit.all_requirements),
-            interpreter_constraints=(
-                partition.interpreter_constraints
-                or PexInterpreterConstraints(bandit.interpreter_constraints)
```

Comment:
The only situation where this could be triggered is if you didn't set `compatibility` anywhere and you went out of your way to set `interpreter_constraints = []`. 

Imo, it's deceptive to have the option because almost always it has no impact on behavior.